libyul/optimiser/Suite.cpp

```diff
@@ -399,7 +399,7 @@ bool OptimiserSuite::isEmptyOptimizerSequence(std::string const& _sequence)
 		default:
 			return false;
 		}
-	return true;
+	return delimiterCount == 1;
```

Comment:
This is fine, but I remember @ekpyron mentioning that we don't really even need this - i.e. checking for whitespace and newlines, as well as the number of semicolons, but rather to simply check if a semicolon delimiter is present in the sequence, and return true if yes, which should then fail later during validation if the sequence is malformed. [Here's](https://github.com/ethereum/solidity/pull/14657#discussion_r1383233923) the comment. Although even there he says he's fine with leaving it as is, so no biggie.

Reply:
Not sure if I follow, do you mean just this?
```c++
bool OptimiserSuite::isEmptyOptimizerSequence(std::string const& _sequence)
{
	return _sequence.find(':') != std::string::npos;
}
```
Also, wouldn't this return the wrong result if you have something like: `"u:"`?

Reply:
No, more like what Daniel mentioned in the above comment (I guess we'd still need to check for white space and newlines: `!return ranges::any_of(_sequence, [](auto _step) { return _step != ':' && _step != ' ' && _step != '\n'; });`

Reply:
AFAIU, when the empty string (or blank spaces only) is input, then the original code (or Daniel's suggested refactor) returns `true`, but the empty string means **no optimizer steps and the default cleanup sequence**. If we want to test if a string is the empty optimizer sequence (both steps and cleanup empty), we must ensure that (at least) one semicolon appears, i.e, ":".

Reply:
I refactored the function to include both the check of number of colons and the one-liner suggested on the previous PR.